hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/multipart/S3MultipartUploadCompleteRequest.java

```diff
@@ -642,10 +661,36 @@ private String multipartUploadedKeyHash(
     StringBuffer keysConcatenated = new StringBuffer();
     for (PartKeyInfo partKeyInfo: partsList) {
       keysConcatenated.append(KeyValueUtil.getFromProtobuf(partKeyInfo
-          .getPartKeyInfo().getMetadataList()).get("ETag"));
+          .getPartKeyInfo().getMetadataList()).get(OzoneConsts.ETAG));
```

Comment:
Regarding the incomplete multipart uploads, the parts that do not have "eTag" yet will return null. In `StringBuffer`, it will append four characters "null". However, I think there is little we can do here, so I think it should be fine to handle the incompatibility.

Reply:
Thank you for the update.